Pirat/Services/ConfigurationService.cs

```diff
@@ -103,11 +105,60 @@ public List<string> GetRegionCodes()
             return this.Regions.Keys.ToList();
         }
 
+        public void ThrowIfNotInLanguage(string regionCode, Offer offer)
+        {
+            foreach (var consumable in offer.consumables)
+            {
+                ThrowIfNotConsumableCategoryInLanguage(regionCode, consumable.category);
+            }
+            foreach (var device in offer.devices)
+            {
+                ThrowIfNotDeviceCategoryInLanguage(regionCode, device.category);
+            }
+            foreach (var personal in offer.personals)
+            {
+                ThrowIfNotPersonnelAreaInLanguage(regionCode, new List<string>(1){personal.area});
+                ThrowIfNotPersonnelQualificationInLanguage(regionCode, new List<string>(1) { personal.qualification });
+            }
+        }
+
+        public void ThrowIfNotConsumableCategoryInLanguage(string regionCode, string category)
+        {
+            if (!Regions[regionCode].Categories.Consumable.Contains(category))
+            {
+                throw new ArgumentException(Codes.FailureCodes.InvalidCategoryConsumable);
+            }
+        }
+
+        public void ThrowIfNotDeviceCategoryInLanguage(string regionCode, string category)
+        {
+            if(!Regions[regionCode].Categories.Device.Contains(category))
+            {
+                throw new ArgumentException(Codes.FailureCodes.InvalidCategoryDevice);
+            }
+        }
+
+        public void ThrowIfNotPersonnelAreaInLanguage(string regionCode, List<string> areas)
+        {
+            if (!areas.Any() || areas.Any(area => !Regions[regionCode].Categories.PersonnelArea.Contains(area)))
```

Comment:
Why is it throwing an error if `areas` is empty? Then, every provided area (which is none) would be defined.

Reply:
Now it throws an error if none, empty or an element is not a valid value